src/app/branded/step-1/branded-checkout-step-1.component.js

```diff
@@ -154,12 +160,177 @@ class BrandedCheckoutStep1Controller {
       }
     }
   }
+
+  loadCart () {
+    this.errorLoadingCart = false
+
+    // Return the observable instead of subscribing here
+    return this.cartService.get().pipe(
+      tap(data => {
+        // Setting cart data and analytics
+        this.cartData = data
+        this.brandedAnalyticsFactory.saveCoverFees(this.orderService.retrieveCoverFeeDecision())
+        this.brandedAnalyticsFactory.saveItem(this.cartData.items[0])
+        this.brandedAnalyticsFactory.addPaymentInfo()
+        this.brandedAnalyticsFactory.reviewOrder()
+        console.log('done loading cart')
+      }),
+      catchError(error => {
+        // Handle errors by setting flag and logging the error
+        this.errorLoadingCart = true
+        this.$log.error('Error loading cart data for branded checkout step 2', error)
+        return Observable.throw(error) // Rethrow the error so the observable chain can handle it
+      })
+    )
+  }
+
+  loadCurrentPayment () {
+    this.loadingCurrentPayment = true
+
+    return this.orderService.getCurrentPayment().pipe(
+      tap(data => {
+        if (!data) {
+          this.$log.error('Error loading current payment info: current payment doesn\'t seem to exist')
+        } else if (data['account-type']) {
+          this.bankAccountPaymentDetails = data
+        } else if (data['card-type']) {
+          this.creditCardPaymentDetails = data
+        } else {
+          this.$log.error('Error loading current payment info: current payment type is unknown')
+        }
+        this.loadingCurrentPayment = false
+      }),
+      catchError(error => {
+        this.loadingCurrentPayment = false
+        this.$log.error('Error loading current payment info', error)
+        return Observable.throw(error) // Propagate error
+      })
+    )
+  }
+
+  checkErrors () {
+    // Then check for errors on the API
+    return this.orderService.checkErrors().do(
+      (data) => {
+        this.needinfoErrors = data
+        console.log('checking errors')
+      })
+      .catch(error => {
+        this.$log.error('Error loading checkErrors', error)
+        return Observable.throw(error)
+      })
+  }
+
+  submitOrderInternal () {
+    this.loadingAndSubmitting = true
+    // Start by loading the cart
+    this.loadCart()
+      .mergeMap(() => {
+        // After loadCart completes, call loadCurrentPayment
+        console.log('loadCurrentPayment')
+        return this.loadCurrentPayment()
+      })
+      .mergeMap(() => {
+        // After loadCurrentPayment completes, call checkErrors
+        console.log('checkErrors')
+        return this.checkErrors()
+      })
+      .mergeMap(() => {
+      // Clear previous submission errors
+        delete this.submissionError
+        delete this.submissionErrorStatus
+
+        // Prevent multiple submissions
+        if (this.submittingOrder) { return Observable.of(null) }
+
+        this.radioStationName = this.orderService.retrieveRadioStationName()
```

Comment:
I think I was wrong about needing this back. As I dig deeper, it looks like this was only used by the review page to display it back to the user, and we have gotten rid of the review step.